### PR TITLE
Fetching review

### PR DESCRIPTION
**Pull request by lihter**
- Created `Author` and `Review` models on every layer (client, ds, repo, use case)
- Added `fillReview` function to `MovieDetailDelegate` that will be called when the reviews are fetched
- Removed function parameter from `init()` in `ReviewView`, now the data will be given through `populate` method
- Added function for fetching reviews to every layer (client, ds, repo, use case)

Simulator Screen Shot - iPhone 12 - 2021-08-04 at 11 42 17

I accidentally committed "Fixing title ..." here. The changes I did in that commit were dedicated to fixing the differences on `DetailView` `titleLabel` as shown in pictures below.

Before:
Simulator Screen Shot - iPhone 12 - 2021-08-04 at 14 42 16

After:
Simulator Screen Shot - iPhone 12 - 2021-08-04 at 14 21 07

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/ViewModels/ReviewViewModel.swift [hunk 1]
```diff
@@ -8,3 +8,18 @@ struct ReviewViewModel {
...
+
+extension ReviewViewModel {
+    
+    init(fromModel model: ReviewModel) {
+        self.init(
+            author: model.author,
+            createdAt: String(model.createdAt
+                                .prefix(10)
+                                .split(separator: "-")
+                                .joined(separator: "/")),
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/ViewModels/ReviewViewModel.swift]: Format correctly

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift [hunk 2]
```diff
@@ -74,6 +74,20 @@ class MoviesUseCase: MoviesUseCaseProtocol {
             self?.mapResult(result: result, completion: completion)
         }
     }
+    
+    func getReview(for movieId: Int, completion: @escaping(Result<ReviewModel, RequestError>) -> Void) {
+        moviesDataRepo.fetchReviews(for: movieId) { result in
+            switch result {
+            case .success(let reviews):
+                if reviews.count > 0 {
```
Comment (marinzg) [inline MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift]: `guard`

**Review by marinzg — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/ViewModels/ReviewViewModel.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/ViewModels/ReviewViewModel.swift]: [Here](https://github.com/lihter/Movie-App/pull/34/commits/7d765b025ebd89ab8c9483577954de04870bccf6#diff-fdb5e671ec0b6b557986ccda300de0696d2237a33e5b1fab3b98d7cd29d2c3fcR15-R23)

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift]: [Here](https://github.com/lihter/Movie-App/pull/34/commits/7d765b025ebd89ab8c9483577954de04870bccf6#diff-88fed2adfc92bb1f90d676d6f9e69ec5d3af18a8361903c9284e6d0e70098470R83)

**Review by marinzg — APPROVED** [review]

**Pull request merged**